### PR TITLE
[hw,i2c,dv] Update descriptions in testplan

### PR DESCRIPTION
**Pull request by crteja**
- Update Stimulus description for target glitch test
- Fixed typos in test names

### CODE REVIEW

**Review by marnovandermaas — APPROVED** [review]
Looks reasonable to me. Let's have one more person take a look at it to double check.

File: hw/ip/i2c/data/i2c_testplan.hjson [hunk 1]
```diff
@@ -457,13 +458,18 @@
...
+
+           Stimulus:
+              - Configure DUT/Agent to Target/Host mode respectively
+              - Issue a new request(RStart) to DUT during an active transfer
+              - Stop current request(Stop) to DUT during an active transfer
+              - Continue issuing requests to check if DUT FSM handles the transaction correctly
 
             Checking:
-            Dut can accept a new transaction after discarding interrupted transaction.
+            Ensure all transactions including glitches are observed in ACQDAT FIFO
```
Comment (hcallahan-lowrisc) [inline hw/ip/i2c/data/i2c_testplan.hjson:472]:
```suggestion
            Ensure all transactions including glitches are observed in i2c.ACQDATA FIFO
```

**Review by hcallahan-lowrisc — APPROVED** [review]
LGTM with minor nit :+1:

File: hw/ip/i2c/data/i2c_testplan.hjson (hunk 1 above)
Comment (crteja) [reply hw/ip/i2c/data/i2c_testplan.hjson]: Updated as per comments in [d7ad31e](https://github.com/lowRISC/opentitan/compare/fd90c0095a271793d33af41ee3cca9481298aa41..d7ad31ebedff7e0f905a8b6298d6909417eb1cff)

**Pull request merged**